tdrs-frontend/src/components/IdleTimer/IdleTimer.jsx

```diff
@@ -0,0 +1,123 @@
+import React, { useRef, useState, useEffect } from 'react'
+import { useIdleTimer } from 'react-idle-timer'
+import { useDispatch } from 'react-redux'
+import Button from '../Button'
+import { fetchAuth } from '../../actions/auth'
+
+/**
+ * IdleTimer uses the `react-idle-timer` package to watch for user inactivity.
+ * There are environment variables set for the timeout time (how long before the
+ * inactivity modal pops up) and the time between when the modal pops up and the
+ * user is automatically logged out.
+ * There is a second variable for the debounce time to minimize calls to
+ * `/v1/auth_check` on user activity.
+ */
+function IdleTimer() {
+  const [isModalVisible, setIsModalVisible] = useState(false)
+  const dispatch = useDispatch()
+
+  useEffect(() => {
+    function keyListener(e) {
+      const listener = keyListenersMap.get(e.keyCode)
+      return listener && listener(e)
+    }
+    document.addEventListener('keydown', keyListener)
+
+    return () => document.removeEventListener('keydown', keyListener)
+  })
+
+  useEffect(() => {
+    let timer
+    if (isModalVisible) {
+      headerRef.current.focus()
+      timer = setTimeout(onSignOut, process.env.REACT_APP_LOGOUT_TIME)
+    }
+
+    return () => clearTimeout(timer)
+  })
```

Comment:
This effect starts a secondary timer when the modal is opened so that a user can be logged out if they make no decision on the modal. This also focuses on the modal's `h1`.